CRM/Core/Permission/Joomla.php

```diff
@@ -131,4 +135,74 @@ public function checkGroupRole($array) {
     return FALSE;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function upgradePermissions($permissions) {
```

Comment:
Just to @ginkgomzd's comments about not explaining the array_flip - it would make sense to document the format of the $permissions array - although presumably in the fn this extends. The array_flip is pretty clear if the array format is understood